### PR TITLE
Add support for 1.19

### PR DESCRIPTION
**Pull request by Moudoux**
1.19 introduces additional key exchanging, and the login process has been modified. This PR outlines a rough prototype on how to use the new system. Key verification is not yet implemented, but can easily be extracted from the Minecraft login packet as it contains the code required to verify.

It also appears necessary to change the client version in the handshake due to the new login protocol changes, or else the client won't connect.

### CODE REVIEW

File: src/server/handshake.js [hunk 1]
```diff
@@ -7,8 +7,14 @@ module.exports = function (client, server, { version, fallbackVersion }) {
     client.serverHost = packet.serverHost
     client.serverPort = packet.serverPort
     client.protocolVersion = packet.protocolVersion
-
+    
+    if (client.protocolVersion >= 759) {
+      client.version = "1.19";
+    }
+    
+    /*
```
Comment (andriycraft) [inline src/server/handshake.js]: Why this got commented?

File: src/server/handshake.js [hunk 2]
```diff
@@ -25,6 +31,7 @@ module.exports = function (client, server, { version, fallbackVersion }) {
     } else if (client.protocolVersion !== server.mcversion.version && packet.nextState !== 1) {
       client.end('Wrong protocol version, expected: ' + server.mcversion.version + ' and you are using: ' + client.protocolVersion)
     }
+    */
```
Comment (andriycraft) [inline src/server/handshake.js]: same here

File: src/version.js [hunk 3]
```diff
@@ -2,5 +2,5 @@
 
 module.exports = {
   defaultVersion: '1.18.2',
```
Comment (andriycraft) [inline src/version.js:4]: mabye change to 1.19?

**Review by andriycraft — CHANGES_REQUESTED** [review]

**Comment by IceTank** [comment]
CI is failing

**Comment by SpencerBurgess** [comment]
> CI is failing

I believe the CI is failing due to the code the author commented out. @Moudoux are you still working on this?

I can take a look over the next few days, getting excited to have the library working with 1.19

**Comment by Moudoux** [comment]
I'm not working on this, I have a working version of the library for personal use, and this PR serves to show the changes needed to be made to the login procedure in order for 1.19 clients to connect. It is not complete (Player key needs to be verified)

**Pull request closed without merging**